app/controllers/checkout_controller.rb

```diff
@@ -0,0 +1,85 @@
+class CheckoutController < ApplicationController
+
+  # Generate session token for payment
+  def generate_session_token
+    @session = PagSeguro::Session.create
+
+    if @session.id
+      render json: @session.id.to_json
+    else
+      render json: @session.errors.to_json
+    end
+  end
+
+  # Send payment data for pagseguro
+  def create
+
+    # Create order in database
+    order = Order.create(user: @current_user, status: "Em análise")
+
+    chopps = [] # Array with pagseguro items
+
+    # Format params[:chopps] -> {chopps: [
+    # {amount: 8.00, quantity: 1, size: 1000, chopp_type: "tradicional", collar: 2},
+    # {amount: 5.00, quantity: 1, size: 500, chopp_type: "vinho", collar: 0}]}
+    chopps_params = params[:chopps]
+    chopps_params.each do |t|
+      # Creating a chopps in database
+      chopp = Chopp.create(order: order, price: t['amount'], size: t['size'],
+                           chopp_type: t['chopp_type'], collar: t['collar'])
+      # Creating pagseguro items
+      chopps << PagSeguro::Item.new(id: chopp.id, description: "Chopp",
+                                    amount: chopp.price ,  quantity: t['quantity'])
+    end
+
+    # Creating payment pagseguro request
+    payment = PagSeguro::CreditCardTransactionRequest.new
+
+    payment.payment_mode = "gateway"
+    payment.notification_url = "https://fast-retreat-18030.herokuapp.com/notification"
+    payment.reference = order.id
+
+
+    chopps.each do |chopp|
+      payment.items << chopp
+    end
+
+    payment.sender = {
+      hash: params[:sender_hash],
+      name: params[:name],
+      email: "c18315567957132943379@sandbox.pagseguro.com.br" # Email must be with this domain because it is in the test environment
```

Comment:
Tente colocar isso no arquivo de configuração do figaro também!

Reply:
Acho que não precisa, o email só está assim por estarmos no ambiente de teste. Qualquer email @sandbox.pagseguro.com.br vai funcionar. No caso quando mudássemos para produção, seria o email de verdade como `email : params[:email]`.

Reply:
Se eu não mudar isso, nas minhas transações do sandbox vão ser exibidas as transações?

Reply:
Esse email é o do comprador @pwener. Quando a transação é registrada, lá no site do pagseguro , mostra os dados do comprador. O email do vendedor é o seu email, que você utilizou pra fazer o cadastro no pagseguro, por exemplo o meu: "victor.hmfd@gmail.com.

O que eu to tentando te explicar é que pro ambiente de testes precisa ser um email com domínio  : @sandbox.pagseguro.com.br. Qualquer email com esse domínio vai funcionar, por exemplo: sauhshuaussauh@sandbox.pagseguro.com.br.